### ISSUE TITLE
Support single-statement trigger bodies on RDBMS that only support SQL triggers

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15347]
Some RDBMS don't support procedural languages in triggers, only SQL statements (and thus, usually, only `AFTER` triggers).

Simple, single-statement trigger bodies can be translated to work on those RDBMS, e.g.:

**Instead of**:

```sql
CREATE TRIGGER T_TRIGGERS_AFTER_TRG_U
AFTER UPDATE ON T_TRIGGERS_AFTER
REFERENCING OLD AS OLD NEW AS NEW
FOR EACH ROW
BEGIN
  UPDATE T_TRIGGERS_AFTER
  SET C = (OLD.C + 1)
  WHERE ID = NEW.ID;
END;
```

**Do this**:

```sql
CREATE TRIGGER T_TRIGGERS_AFTER_TRG_U
AFTER UPDATE ON T_TRIGGERS_AFTER
REFERENCING OLD AS OLD NEW AS NEW
FOR EACH ROW
UPDATE T_TRIGGERS_AFTER
SET C = (OLD.C + 1)
WHERE ID = NEW.ID;
```

Affected dialects include:

- [x] Derby
- [x] SQLite

Features to emulate:

- `WHEN` (instead of an `IF` statement, we'll have to patch the query): See https://github.com/jOOQ/jOOQ/issues/15355

**Comment by lukaseder** [comment]
This means, we can now add support for `CREATE TRIGGER` in these dialects!

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.19.

I'll add more dialects to the list once I discover them.
